### DISCUSSION TITLE
Chapter 11 - Example of Batching Requests with callbacks and how to do error handling?

### DISCUSSION THREAD

**Discussion by dannyreg** [PacktPublishing/Node.js-Design-Patterns-Third-Edition#46] [General]
I was wondering if you could perhaps provide an example of batching requests with callbacks? Or if it's even possible with callbacks? I ask cuz I was trying to implement this in a node GRPC server I have but since it's not a Promise based implementation, I have to use callbacks. 

[I found some things online about using a queue of callbacks,](https://www.alxolr.com/articles/how-to-serve-10x-more-request-using-asynchronous-request-batching) but I am not sure if this considered batching? 

Also what should happen if one of the requests contains an error? Will a simple catch send it to the appropriate requests.

**Comment by mariocasciaro** [comment]
Hi @HappyZombies , you can find an example in the code samples of the previous edition of the book. Please find the link below.

**Spoiler alert for other readers of the book**. This is essentially a possible answer to one of the exercises in Chapter 11. So if you still want to try to solve the exercise by yourself, don't look at the file below.

https://github.com/PacktPublishing/Node.js_Design_Patterns_Second_Edition_Code/blob/master/Chapter09/03_batching_and_caching/totalSalesBatch.js

**Comment by dannyreg** [comment]
Ah so it does use queues! Makes sense then. Thanks for answering -- btw this book is incredible! Thank you so much for taking the time to write it.

**Comment by mariocasciaro** [comment]
Thank you ☺️
